From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuvraj Saxena <ysaxenax@gmail.com>
Date: Thu, 17 Sep 2026 19:52:31 +0530
Subject: [PATCH] 9648: ItemSerializer: include Central Directory /
 End-of-Central Directory metadata when serializing folder item

ZipOutputStream.closeEntry() does not automatically invoke ZipOutputStream.finish(). As a result, the serialized ZIP data does not include the Central Directory metadata because the ZIP structure is never finalized during serialization. This change fixes the issue by ensuring the ZIP stream is properly finalized before serialization.

Fixes https://github.com/NationalSecurityAgency/ghidra/issues/9649

Signed-off-by: Yuvraj Saxena <ysaxenax@gmail.com>
---
 .../main/java/ghidra/framework/store/local/ItemSerializer.java   | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Ghidra/Framework/FileSystem/src/main/java/ghidra/framework/store/local/ItemSerializer.java b/Ghidra/Framework/FileSystem/src/main/java/ghidra/framework/store/local/ItemSerializer.java
index f6a71e4682..fcda5dbe40 100644
--- a/Ghidra/Framework/FileSystem/src/main/java/ghidra/framework/store/local/ItemSerializer.java
+++ b/Ghidra/Framework/FileSystem/src/main/java/ghidra/framework/store/local/ItemSerializer.java
@@ -109,6 +109,7 @@ public class ItemSerializer {
 			itemOut.flush();
 
 			zipOut.closeEntry();
+			zipOut.finish();
 			zipOut.flush();
 			success = true;
 		}
-- 
2.45.1

